### ISSUE TITLE
Public API comparison between v2.4.1 and v2.5 release branch

### ISSUE THREAD

**Issue by sbrugman** [pytorch/pytorch#137412]
Below if a comparison of the Public API between v2.4.1 and the v2.5 release branch (https://github.com/pytorch/pytorch/issues/135522) @kit1980 
Most changes are not that exciting, but some might be helpful for writing the release notes.

Removed parameters:

- torch/ao/quantization/observer.py:984: HistogramObserver.__init__: `upsample_rate` removed (set to default of 16)
- torch/autograd/profiler.py:197: profile.__init__: `use_mtia` removed
- torch/fx/experimental/symbolic_shapes.py:4387: ShapeEnv.get_implications: `compute_hint` removed
- torch/fx/experimental/symbolic_shapes.py:5117: ShapeEnv.evaluate_expr: `expect_rational` removed
- torch/export/exported_program.py:663: ExportedProgram.__init__: `verifier` and `tensor_constants` removed

Required parameter added:

- torch/distributed/elastic/rendezvous/etcd_rendezvous.py:152: EtcdRendezvousHandler.__init__: `local_addr`  was added as required (but can be set to `None`). Perhaps set to `None` by default?

Renamed parameters:

- torch/__init__.py:970: typename(o): Parameter `o` was renamed to `obj`
- torch/masked/maskedtensor/core.py:18: is_masked_tensor: Parameter `a` was renamed to `obj`
- torch/autograd/graph.py:172: Node.__subclasshook__: Parameter `C` was renamed to `subclass`

Parameter was changed to positional-only:

- torch/__init__.py:1005: is_tensor(obj)
- torch/__init__.py:1025: is_storage(obj)
- torch/__init__.py:1117: set_default_tensor_type(t)
- torch/__init__.py:1490: set_warn_always(b)
- torch/_tensor.py:1165: Tensor.__contains__(self)
- torch/_tensor.py:1165: Tensor.__contains__(element)

Parameter was changed to keyword-only:

- torch/onnx/__init__.py:137: export(export_params):
- torch/onnx/__init__.py:137: export(verbose)
- torch/onnx/__init__.py:137: export(training)
- torch/onnx/__init__.py:137: export(input_names)
- torch/onnx/__init__.py:137: export(output_names)
- torch/onnx/__init__.py:137: export(operator_export_type)
- torch/onnx/__init__.py:137: export(opset_version)
- torch/onnx/__init__.py:137: export(do_constant_folding)
- torch/onnx/__init__.py:137: export(dynamic_axes)
- torch/onnx/__init__.py:137: export(keep_initializers_as_inputs)
- torch/onnx/__init__.py:137: export(custom_opsets)
- torch/onnx/__init__.py:137: export(export_modules_as_functions)
- torch/onnx/__init__.py:137: export(autograd_inlining)
- torch/onnx/__init__.py:137: export(dynamo)

Removed public objects:

- torch/onnx/__init__.py:0: ONNXProgramSerializer
- torch/onnx/__init__.py:0: InvalidExportOptionsError
- torch/onnx/symbolic_helper.py:0: is_caffe2_aten_fallback
- torch/onnx/errors.py:0: CheckerError
- torch/distributed/distributed_c10d.py:0: ProcessGroupCudaP2P
- torch/distributed/tensor/parallel/style.py:0: SequenceParallel.sequence_dim
- torch/distributed/__init__.py:0: ProcessGroupCudaP2P
- torch/export/__init__.py:0: dynamic_dim
- torch/export/dynamic_shapes.py:0: dynamic_dim

**Comment by kit1980** [comment]
Thanks!

**Comment by albanD** [comment]
very cool!
Not sure how this was generated but from checking "torch/init.py:970: typename(o): Parameter o was renamed to obj", this one is ok I think as it is a positional-only argument. So shouldn't be able to impact the user right?

**Comment by sbrugman** [comment]
> very cool! Not sure how this was generated but from checking "torch/init.py:970: typename(o): Parameter o was renamed to obj", this one is ok I think as it is a positional-only argument. So shouldn't be able to impact the user right?

Absolutely right, that change won't impact users at all.

**Comment by malfet** [comment]
Flagging to triage review, to discuss whether one should consider changing something to positional only argument ( Python-3.8 feature) a BC breaking behavior
